File: codeforces/1203/c.py
import math
n = int(input())
c = list(map(int,input().split()))
g = 0
for i in c:
	g = math.gcd(g,i)
sq = (int)math.sqrt(g)
ans = 0
for i in range(1,sq+1):
	if(g%i!=0):
		continue
	elif i==sq:
		ans += 1
	else:
		ans += 2
print(ans)

File: codeforces/1203/c.cpp
#include<bits/stdc++.h>
using namespace std;

#define pb push_back
#define ll long long
#define maxn 100005
#define fr(i,j,k) for(int i=j;i<k;i++)
#define f(n) fr(i,0,n)
#define f1(n) fr(i,1,n+1)
#define ms(i) memset(i,0,sizeof(i));
#define ms1(i) memset(i,-1,sizeof(i));
#define F first
#define S second

int main(){
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	int n;
	while(cin >> n){
		ll gcd = 0;
		f(n){
			ll add;
			cin >> add;
			gcd = __gcd(gcd,add);
		}
		int ans = 0;
		for(ll i=1;i<=sqrt(gcd);i++){
			if(gcd%i)continue;
			if(i*i==gcd)ans++;
			else ans+=2;
		}
		cout << ans << endl;
	}
}